### PR TITLE
Update rn-integrate script to more aggressively update align dependencies

### PR DESCRIPTION
**Pull request by acoates-ms**
Various dependencies throughout the repo are stale, which can cause issues within the repo.

For example, the version of @react-native/babel-preset was not correctly triggering @react-native/babel-plugin-codegen due to it being stale.

I've updated the rn-integrate script to more aggressively update various dependencies throughout the repo. This reduces the duplication in yarn.lock and generally should help us keep more up to date with core.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12740)

### CODE REVIEW

File: packages/e2e-test-app-fabric/jest.config.js [hunk 1]
```diff
@@ -10,11 +10,6 @@
 // https://jestjs.io/docs/en/configuration.html
```
Comment (tido64) [inline packages/e2e-test-app-fabric/jest.config.js:10]: Should we consolidate `jest.config.js` and `jest.setup.js`? They look mostly the same to me.

File: yarn.lock [hunk 2]
```diff
@@ -2386,6 +2365,30 @@
   dependencies:
     joi "^17.2.1"
 
+"@react-native-community/cli@12.3.2":
```
Comment (tido64) [inline yarn.lock:2368]: It looks like we ended up with both `@react-native-community/cli` 12.x and 13.x here.

File: yarn.lock [hunk 3]
```diff
@@ -2415,6 +2418,11 @@
   resolved "https://registry.yarnpkg.com/@react-native-picker/picker/-/picker-2.6.1.tgz#3b20ddd1385fab0487db103dc6519570f8892e6d"
   integrity sha512-oJftvmLOj6Y6/bF4kPcK6L83yNBALGmqNYugf94BzP0FQGpHBwimVN2ygqkQ2Sn2ZU3pGUZMs0jV6+Gku2GyYg==
 
+"@react-native/assets-registry@0.73.1":
```
Comment (tido64) [inline yarn.lock:2421]: It also looks like we introduced dupes of `@react-native/assets-registry` and other packages.

File: yarn.lock (hunk 2 above)
Comment (acoates-ms) [reply yarn.lock:2368]: Ah good catch, through react-native-xaml which still depends on @types/react-native, which brings in react-native * as a dep.  

https://github.com/microsoft/react-native-xaml/pull/279 will remove the now unnecessary @types/react-native dependency from react-native-xaml, then we can stop bringing in multiple react-native versions.

**Review by chiaramooney — APPROVED** [review]
👍

**Pull request merged**